ingest/src/main/java/com/connexta/ingest/adaptors/S3Adaptor.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.ingest.adaptors;
+
+import com.connexta.ingest.service.api.IngestRequest;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.UUID;
+import javax.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+@Service
+public class S3Adaptor implements Adaptor {
+
+  private S3Client s3Client;
+
+  // We'll probably change these when we move to a cloud friendly configuration
+  private static final String S3_BUCKET = "staged-products";
+  private static final String S3_ENDPOINT = "http://127.0.0.1:9000";
```

Comment:
If you keep the properties file, you should be able to do something like this in Spring:
```
@Value("${s3.endpoint}")
private string S3endpoint;
```

To populate it from the config file.

Reply:
I had that originally and I can't remember why I changed it. I think we may be able to override those properties with env variables or similar when running in docker.

Reply:
yes if you have the `@Value("${s3.endpoint}")` in your Configuration you can override it via `application.yml`, `ENV` or `--s3.endpoint=<stuff>`

Reply:
I moved the S3 configuration over to the Docker side of things.